src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1529,6 +1529,12 @@ def _process_type_spec(self, parent, type_spec):
 
         elif isinstance(type_spec, Fortran2003.Declaration_Type_Spec):
             # This is a variable of derived type
+            if type_spec.children[0].lower() == "class":
```

Comment:
Maybe there aren't any other cases but could we do `not type_spec.children[0].lower() == "type"` here just in case?

Reply:
Good idea. Done.